src/librustc/infer/lexical_region_resolve/graphviz.rs

```diff
@@ -187,12 +184,10 @@ impl<'a, 'gcx, 'tcx> dot::Labeller<'a> for ConstraintGraph<'a, 'gcx, 'tcx> {
             None => bug!("no node_id found for node: {:?}", n),
         };
         let name = || format!("node_{}", node_id);
-        match dot::Id::new(name()) {
-            Ok(id) => id,
-            Err(_) => {
-                bug!("failed to create graphviz node identified by {}", name());
-            }
-        }
+
+        dot::Id::new(name()).unwrap_or_else(|_|
+            bug!("failed to create graphviz node identified by {}", name())
+        )
```

Comment:
Tweak the formatting here, as it makes it look like the closing parenthesis is for the `unwrap_or_else`, while it is for `Id::new`.